android/quest/src/androidTest/java/org/smartregister/fhircore/performance/HouseholdRegisterPerformanceTests.kt

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2021-2023 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.performance
+
+import androidx.benchmark.junit4.BenchmarkRule
+import androidx.benchmark.junit4.measureRepeated
+import androidx.test.platform.app.InstrumentationRegistry
+import dagger.hilt.android.testing.HiltAndroidRule
+import dagger.hilt.android.testing.HiltAndroidTest
+import java.io.File
+import java.nio.file.Files
+import javax.inject.Inject
+import kotlin.io.path.Path
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.smartregister.fhircore.engine.data.local.DefaultRepository
+import org.smartregister.fhircore.engine.data.local.register.RegisterRepository
+import timber.log.Timber
+
+@HiltAndroidTest
+class HouseholdRegisterPerformanceTests {
+
+  @get:Rule val benchmarkRule = BenchmarkRule()
+
+  @Inject lateinit var defaultRepository: DefaultRepository
+
+  @Inject lateinit var registerRepository: RegisterRepository
+
+  @get:Rule val hiltRule = HiltAndroidRule(this)
+
+  @Before
+  fun setUp() {
+    hiltRule.inject()
+
+    if (InstrumentationRegistry.getInstrumentation().context.externalCacheDir == null) {
+      Timber.e("Instrumentation registry is null")
+    } else {
+      Timber.e("Instrumentation registry is NOT NULL")
+    }
+    // Might need to login
+
+    runBlocking {
+      defaultRepository.configurationRegistry.loadConfigurations(
+        "app/debug",
+        InstrumentationRegistry.getInstrumentation().targetContext,
+      ) { loadConfigSuccessful ->
+      }
+    }
+
+    // Copy over the db
+    val appContext = InstrumentationRegistry.getInstrumentation().targetContext
+    val testContext = InstrumentationRegistry.getInstrumentation().context
+
+    val resourcesDbInputStream = testContext.assets.open("resources.db")
+
+    // Delete the database files
+    "/data/data/${appContext.packageName}/databases/resources.db".deleteFileIsExists()
+    "/data/data/${appContext.packageName}/databases/resources.db-shm".deleteFileIsExists()
+    "/data/data/${appContext.packageName}/databases/resources.db-wal".deleteFileIsExists()
+
+    // Copy over the db
+    Files.copy(
+      resourcesDbInputStream,
+      Path("/data/data/${appContext.packageName}/databases/resources.db"),
+    )
+  }
+
+  @Test
+  fun benchmarkPage0() {
+    benchmarkingFunctionality(0)
```

Comment:
can this helper function return the time it took? then in here we can put it in an assert, like
```suggestion
    val secondsToRun = benchmarkingFunctionality(0)
    assertTrue(secondsToRun < 180)
```
or if these all take around the same time to run, we can put the assert in the helper function

Reply:
The benchmarks don't return the timing results. The timings results are saved in a file. These are provided as median, max, min and all the individual iteration runtimes. 

For us to assert, we need to run a gradle task to parse this file and compare the times for each test result

Reply:
cool